src/modules/parser.ts

```diff
@@ -221,14 +221,18 @@ function checkArgs (argsList: QueryType['operation']['args'], operationArg: stri
 
 /**
  * Parses the operation bit of the query
- * @param {queryType} query The JSON-Like query to be parsed
+ * @param {QueryType} query The JSON-Like query to be parsed
+ * @param {boolean} allowEmptyFields If 'true', empty fields are allowed (which is not considered a good practice)
  * @return {string} Parsed operation query
  */
-function parseOperation (query: QueryType): string {
+function parseOperation (query: QueryType, allowEmptyFields: boolean): string {
```

Comment:
The problem here is that you add a new **required** parameter. This introduces a breaking change. So my suggestion would be this:

```suggestion
function parseOperation (query: QueryType, allowEmptyFields: boolean = false): string {
```

Reply:
had the same thought initially, although it is kind of "private" function...so I decided against